gravitee-apim-console-webui/src/management/api/portal/user-group-access/groups/api-portal-groups.component.spec.ts

```diff
@@ -0,0 +1,200 @@
+      expect(selectedOptions.length).toEqual(1);
+      expect(await selectedOptions[0].getText()).toEqual('My Group');
+
+      await selection.clickOptions({ text: 'New Group' });
+      expect(await selection.getValueText()).toEqual('My Group, New Group');
+      await selection.close();
+
+      expect(await fillForm.isControlDirty()).toEqual(true);
+
+      const saveBar = await loader.getHarness(GioSaveBarHarness);
+      expect(await saveBar.isVisible()).toEqual(true);
+      await saveBar.clickSubmit();
+
+      expectOneApiGet(fakeApi({ id: API_ID, groups: ['my_group'] }));
+      expectOneApiPut(fakeApi({ id: API_ID, groups: ['my_group', 'new_group'] }));
+
+      // expect reloaded component
+      expectOneGroupList([fakeGroup({ id: 'my_group', name: 'My Group' }), fakeGroup({ id: 'new_group', name: 'New Group' })]);
+      expectOneApiGet(fakeApi({ id: API_ID, groups: ['my_group', 'new_group'] }));
+
+      expect(await fillForm.isControlDirty()).toEqual(false);
+      await selection.open();
+      const newSelectedOptions = await selection.getOptions({ isSelected: true });
+      expect(newSelectedOptions.length).toEqual(2);
+
+      expect(await saveBar.isVisible()).toEqual(false);
+    });
+
+    it("should reset to user's original groups after clicking reset button", async () => {
+      expectOneGroupList([fakeGroup({ id: 'my_group', name: 'My Group' }), fakeGroup({ id: 'new_group', name: 'New Group' })]);
+      expectOneApiGet(fakeApi({ id: API_ID, groups: ['my_group'] }));
+
+      const selection = await loader.getHarness(MatSelectHarness);
+      await selection.open();
+      let selectedOptions = await selection.getOptions({ isSelected: true });
+      expect(selectedOptions.length).toEqual(1);
+
+      await selection.clickOptions({ text: 'New Group' });
+      selectedOptions = await selection.getOptions({ isSelected: true });
+      expect(selectedOptions.length).toEqual(2);
+      expect(await selection.getValueText()).toEqual('My Group, New Group');
+      await selection.close();
+
+      const saveBar = await loader.getHarness(GioSaveBarHarness);
+      expect(await saveBar.isResetButtonVisible()).toEqual(true);
+      await saveBar.clickReset();
+
+      await selection.open();
+      selectedOptions = await selection.getOptions({ isSelected: true });
+      expect(selectedOptions.length).toEqual(1);
+
+      expect(await saveBar.isVisible()).toEqual(false);
+    });
+  });
+
+  describe('Groups tab for user with read-only rights', () => {
+    beforeEach(async () => {
+      await init(['api-definition-r', 'api-gateway_definition-u']);
+    });
+
+    it('should display list of groups', async () => {
+      expectOneGroupList([fakeGroup({ id: 'my_group', name: 'My Group' }), fakeGroup({ id: 'my_other_group', name: 'My Other Group' })]);
+      expectOneApiGet(fakeApi({ id: API_ID, groups: ['my_group', 'my_other_group'] }));
+
+      expect(fixture.nativeElement.innerHTML.includes('My Group, My Other Group'));
```

Comment:
suggestion: we could use a css class to build a more precise query and avoid inspecting the whole component dom